lib/model/store.dart

```diff
@@ -387,112 +387,110 @@ class PerAccountStore extends ChangeNotifier with StreamStore, MessageStore {
-      notifyListeners();
-    } else if (event is AlertWordsEvent) {
-      assert(debugLog("server event: alert_words"));
-      // We don't yet store this data, so there's nothing to update.
-    } else if (event is UserSettingsUpdateEvent) {
-      assert(debugLog("server event: user_settings/update ${event.property?.name ?? '[unrecognized]'}"));
-      if (event.property == null) {
-        // unrecognized setting; do nothing
-        return;
-      }
-      switch (event.property!) {
-        case UserSettingName.twentyFourHourTime:
-          userSettings?.twentyFourHourTime        = event.value as bool;
-        case UserSettingName.displayEmojiReactionUsers:
-          userSettings?.displayEmojiReactionUsers = event.value as bool;
-        case UserSettingName.emojiset:
-          userSettings?.emojiset                  = event.value as Emojiset;
-      }
-      notifyListeners();
-    } else if (event is CustomProfileFieldsEvent) {
-      assert(debugLog("server event: custom_profile_fields"));
-      customProfileFields = _sortCustomProfileFields(event.fields);
-      notifyListeners();
-    } else if (event is RealmUserAddEvent) {
-      assert(debugLog("server event: realm_user/add"));
-      users[event.person.userId] = event.person;
-      notifyListeners();
-    } else if (event is RealmUserRemoveEvent) {
-      assert(debugLog("server event: realm_user/remove"));
-      users.remove(event.userId);
-      autocompleteViewManager.handleRealmUserRemoveEvent(event);
-      notifyListeners();
-    } else if (event is RealmUserUpdateEvent) {
-      assert(debugLog("server event: realm_user/update"));
-      final user = users[event.userId];
-      if (user == null) {
-        return; // TODO log
-      }
-      if (event.fullName != null)       user.fullName       = event.fullName!;
-      if (event.avatarUrl != null)      user.avatarUrl      = event.avatarUrl!;
-      if (event.avatarVersion != null)  user.avatarVersion  = event.avatarVersion!;
-      if (event.timezone != null)       user.timezone       = event.timezone!;
-      if (event.botOwnerId != null)     user.botOwnerId     = event.botOwnerId!;
-      if (event.role != null)           user.role           = event.role!;
-      if (event.isBillingAdmin != null) user.isBillingAdmin = event.isBillingAdmin!;
-      if (event.deliveryEmail != null)  user.deliveryEmail  = event.deliveryEmail!.value;
-      if (event.newEmail != null)       user.email          = event.newEmail!;
-      if (event.customProfileField != null) {
-        final profileData = (user.profileData ??= {});
-        final update = event.customProfileField!;
-        if (update.value != null) {
-          profileData[update.id] = ProfileFieldUserData(value: update.value!, renderedValue: update.renderedValue);
-        } else {
-          profileData.remove(update.id);
+    switch(event) {
+      case HeartbeatEvent():
+        assert(debugLog("server event: heartbeat"));
+      case RealmEmojiUpdateEvent():
```

Comment:
nit: let's separate these cases visually from each other with blank lines:
```suggestion
        assert(debugLog("server event: heartbeat"));

      case RealmEmojiUpdateEvent():
```

I think that'll help a bit in visually scanning to find where a given event is handled.

(With the `if` form, the `is FooEvent` stuck out more to the right, and the close-brace also made a more distinctive marker at the left, so there was less need for extra visual emphasis.)